### DISCUSSION TITLE
Add support for storing and loading FTS index

### DISCUSSION THREAD

**Discussion by Raidus** [duckdb/duckdb#4029] [Ideas]
Hey 👋

Would be great if you could add the ability to store and load FTS index. Currently I have the create the FTS index on each application startup which takes in total 14s for two indexes.

**Comment by lnkuiper** [comment]
The FTS index is stored in plain tables/macros in a separate schema. These can be loaded/stored like any other table (in the single-file datatabase storage).

It can of course also be exported using `EXPORT DATABASE ...` Documentation for export can be found here:
https://duckdb.org/docs/sql/statements/export

If exporting, I would suggest using `FORMAT PARQUET`. Then, importing/exporting these parquet files should be much faster than re-computing the index.

**Reply by Raidus** [reply]
Cool thanks! Haven't thought about that! I tried your suggestion but somehow the bm25 function does not exist. I don't know if its a bug or I did something wrong. 

Here is a simple example

```python
import duckdb

con = duckdb.connect(":memory:")

q = con.execute("INSTALL 'fts'; LOAD 'fts'")


def export_db():
    con.execute("""CREATE TABLE documents(doc_id VARCHAR, text_content VARCHAR);""")

    con.execute(
        """
    INSERT INTO documents
    VALUES ('doc1', 'face wash wipes'),
        ('doc2', 'Cant you go wrong with it?')
    """
    )

    con.execute(
        """PRAGMA create_fts_index('documents', 'doc_id', 'text_content', stopwords='none',ignore='(\\.|[^a-z])+');"""
    )

    con.execute(f"EXPORT DATABASE './tmp/exported_db' (FORMAT PARQUET)")


def import_db():
    con.execute(f"IMPORT DATABASE './tmp/exported_db'")


def test():

    q = con.execute("PRAGMA show_tables").fetchall()
    print(q)

    sql = f"""
    SELECT text_content, score
    FROM (SELECT *, fts_main_documents.match_bm25(doc_id, 'face', conjunctive:=1) AS score
            FROM documents) sq
    WHERE score IS NOT NULL
    ORDER BY score DESC
    """
    q = con.execute(sql).fetchall()
    print(q)


if __name__ == "__main__":
    # export_db()

    import_db()

    test()

```

**Reply by Raidus** [reply]
I have created the issue https://github.com/duckdb/duckdb/issues/4047
